dotcom-rendering/src/model/enhanceBlocks.ts

```diff
@@ -61,8 +78,13 @@ class BlockEnhancer {
 // IMPORTANT: the ordering of the enhancer is IMPORTANT to keep in mind
 // example: enhanceInteractiveContentElements needs to be before enhanceNumberedLists
 // as they both effect SubheadingBlockElement
-export const enhanceBlocks = (blocks: Block[], format: CAPIFormat): Block[] => {
-	return new BlockEnhancer(blocks, format)
+export const enhanceBlocks = (
+	blocks: Block[],
+	format: CAPIFormat,
+	promotedNewsletter?: Newsletter,
+): Block[] => {
+	return new BlockEnhancer(blocks, format, promotedNewsletter)
+		.insertNewsletterSignup()
```

Comment:
I think actually we need to do this last because enhancing the dividers rely on the initial order of blocks. So it could cause issues if the newsletter embed is inserted between the dinkus and its following paragraph. I think. Best to be safe